static/scripts/pay.ts

```diff
@@ -30,11 +39,22 @@ const withdraw = async (signer: JsonRpcSigner, txData: TxType, predefined: strin
 };
 
 const fetchTreasury = async (): Promise<{ balance: number; allowance: number }> => {
-  const tokenAddress = txData.permit.permitted.token;
-  const tokenContract = new ethers.Contract(tokenAddress, daiAbi, new ethers.providers.Web3Provider((window as any).ethereum));
-  const balance = await tokenContract.balanceOf(txData.owner);
-  const allowance = await tokenContract.allowance(txData.owner, permit2Address);
-  return { balance, allowance };
+  try {
+    const provider = new ethers.providers.Web3Provider((window as any).ethereum)
```

Comment:
You should use a window object polyfill type. I thought we banned `any` in our repository!

For example

```typescript
declare global {
	interface Window {
		ethereum: MetaMaskInpageProvider;
	}
}
```